### PR TITLE
[Merged by Bors] - hdfs getting-started

### LINKED ISSUE #220: Docs: Getting started guide
Part of this epic: https://github.com/stackabletech/documentation/issues/237

Acceptance criteria:
- [ ] a new module `getting_started` exists, with an `index.adoc`, a `installation.adoc` and `first_steps.adoc`
- [ ] The old `installation.adoc` is removed.
- [ ] All the `yaml` and shell snippets are in the `examples/code` directory and can be executed as a script, to test the documentation
- [ ] versions to install are templated using the `template_docs.sh` script and the `templating_vars.yaml` file.
- [ ] Any example files that are incorporated into the getting started guide are removed from the examples directory (because there they are not tested, with the new script they will be tested).

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Getting-started docs.
Closes #220.

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/getting_started/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+= Getting started
+
+This guide will get you started with HDFS using the Stackable Operator. It will guide you through the installation of the Operator and its dependencies, setting up your first HDFS cluster and ...
```
Comment (fhennig) [inline docs/modules/getting_started/pages/index.adoc]: and? :smile:

File: docs/modules/getting_started/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+== Prerequisites
+
+You will need:
+
+* a Kubernetes cluster
+* kubectl
+* Helm
```
Comment (fhennig) [inline docs/modules/getting_started/pages/index.adoc]:
```suggestion
* optional: Helm
```

File: docs/modules/getting_started/pages/first_steps.adoc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+== Setup
+
+=== Zookeeper
+
+To deploy a Zookeeper cluster create one file called `zk.yaml`:
+
+[source,yaml]
+include::example$code/zk.yaml[]
+
+We also need to define a `ZNode` that will be used by the HDFS cluster to reference Zookeeper. Create another file called `znode.yaml`:
```
Comment (fhennig) [inline docs/modules/getting_started/pages/first_steps.adoc]:
```suggestion
We also need to define a ZNode that will be used by the HDFS cluster to reference Zookeeper. Create another file called `znode.yaml`:
```

We don't use this styling on resource kinds.

File: docs/modules/getting_started/pages/first_steps.adoc [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+include::example$code/getting_started.sh[tag=install-zk]
+
+The state of the Zookeeper cluster can be tracked with `kubectl`:
+
+[source]
+include::example$code/getting_started.sh[tag=watch-zk-rollout]
+
+=== HDFS
+
+An HDFS cluster has three components: then `namenode`, the `datanode` and the `journalnode`. Create a file named `hdfs.yaml` defining 2 `namenodes` and one `datanode` and `journalnode` each:
```
Comment (fhennig) [inline docs/modules/getting_started/pages/first_steps.adoc]:
```suggestion
An HDFS cluster has three components: the `namenode`, the `datanode` and the `journalnode`. Create a file named `hdfs.yaml` defining 2 `namenodes` and one `datanode` and `journalnode` each:
```

File: docs/modules/getting_started/examples/code/zk.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: simple-zk
+spec:
+  version: 3.8.0-stackable0.7.1
+  servers:
+    roleGroups:
+      default:
+        selector:
+          matchLabels:
+            kubernetes.io/os: linux
```
Comment (fhennig) [inline docs/modules/getting_started/examples/code/zk.yaml]:
```suggestion
```

File: docs/modules/getting_started/examples/code/zk.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        replicas: 1
+        config: {}
```
Comment (fhennig) [inline docs/modules/getting_started/examples/code/zk.yaml]:
```suggestion
```

File: docs/modules/getting_started/examples/code/hdfs.yaml [hunk 7]
```diff
@@ -81,4 +58,4 @@ spec:
         selector:
           matchLabels:
             kubernetes.io/os: linux
```
Comment (fhennig) [inline docs/modules/getting_started/examples/code/hdfs.yaml]:
```suggestion
```

File: docs/modules/getting_started/examples/code/hdfs.yaml [hunk 8]
```diff
@@ -67,7 +44,7 @@ spec:
...
```
Comment (fhennig) [inline docs/modules/getting_started/examples/code/hdfs.yaml]:
```suggestion
```

Comment (fhennig) [reply docs/modules/getting_started/examples/code/hdfs.yaml]: There's another selector that I can't suggest an edit on that can also be removed (line 32)

File: docs/modules/getting_started/examples/code/hdfs.yaml [hunk 9]
```diff
@@ -1,27 +1,4 @@
...
-apiVersion: zookeeper.stackable.tech/v1alpha1
-kind: ZookeeperZnode
-metadata:
-  name: simple-hdfs-znode
-spec:
-  clusterRef:
-    name: simple-zk
----
 apiVersion: hdfs.stackable.tech/v1alpha1
 kind: HdfsCluster
```
Comment (fhennig) [inline docs/modules/getting_started/examples/code/hdfs.yaml:3]: I think the resource settings (capacity, memory) can probably be left out? If so I'd omit them to make the example as simple as possible.

**Review by fhennig — CHANGES_REQUESTED** [review]
I really like the file copying!

File: docs/modules/getting_started/examples/code/hdfs.yaml (hunk 9 above)
Comment (adwk67) [reply docs/modules/getting_started/examples/code/hdfs.yaml:3]: Done.

File: docs/modules/getting_started/pages/index.adoc (hunk 1 above)
Comment (adwk67) [reply docs/modules/getting_started/pages/index.adoc]: ...suspense :)

Comment (adwk67) [reply docs/modules/getting_started/pages/index.adoc]: Done

**Review by fhennig — APPROVED** [review]

**Comment by adwk67** [comment]
bors merge

**Comment by adwk67** [comment]
bors merge

**Comment by adwk67** [comment]
bors cancel

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**